backend/src/signer/signer.rs

```diff
@@ -0,0 +1,122 @@
+use crate::contracts::generated::summa_contract::Summa;
+use ethers::{
+    abi::{encode, Token},
+    prelude::SignerMiddleware,
+    providers::{Http, Middleware, Provider},
+    signers::{LocalWallet, Signer, WalletError},
+    types::{Address, Signature},
+    utils::keccak256,
+};
+use futures::future::join_all;
+use serde_json::Value;
+use std::{error::Error, fs::File, io::BufReader, path::Path, str::FromStr, sync::Arc};
+
+#[derive(Debug)]
+pub struct SummaSigner {
+    signing_wallets: Vec<LocalWallet>,
+    summa_contract: Summa<SignerMiddleware<Provider<Http>, LocalWallet>>,
+}
+
+impl SummaSigner {
+    pub fn new(
+        private_keys: &[&str],
```

Comment:
Can you specify in the comments what's the difference between `private_keys` and `main_signer_key`